src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -153,6 +153,12 @@ void RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::Predict(
     ResetDeterministic();
   }
 
+  arma::mat resultsTemp;
+  Forward(std::move(arma::mat(predictors.slice(0).colptr(0),
```

Comment:
What about we move this into the for loop and set the site of `result` at the first step?

Reply:
@zoq  that's what I did in the [previous commit](https://github.com/mlpack/mlpack/pull/1847/commits/dc3cbf46d95e9a940c6a4510f6ff45e4e54c63c7).

Reply:
Ahh, right, I think in this case it's faster to check this at the first step once, instead of doing a complete forward pass; or what about we start with ``begin = 1`` (do the first step outside of the for loop). Let me know what you think.

Reply:
Yes, that might work. Thanks @zoq